projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMAddressCompareNode.java

```diff
@@ -51,6 +51,22 @@ public boolean executeI1(TruffleObject val1, TruffleObject val2) {
             return val1 == val2;
         }
 
+        @SuppressWarnings("unused")
+        @Specialization(guards = "isCNull(val1)")
+        public boolean executeI1(LLVMAddress val1, TruffleObject val2) {
+            return false;
+        }
+
+        @SuppressWarnings("unused")
+        @Specialization(guards = "isCNull(val2)")
+
+        public boolean executeI1(TruffleObject val1, LLVMAddress val2) {
+            return false;
+        }
+
+        public static boolean isCNull(LLVMAddress val) {
```

Comment:
I originally just had it as false, but really the only comparison that should be legal is a comparison against NULL. Anything else is at best suspicious, so I thought it better to catch that case and deal with it if an when it arises.